src/cmd/mapper/mapper-upload.go

```diff
@@ -0,0 +1,45 @@
+package mapper
+
+import (
+	"context"
+	"github.com/otterize/otterize-cli/src/pkg/config"
+	"github.com/otterize/otterize-cli/src/pkg/intents"
+	"github.com/otterize/otterize-cli/src/pkg/mapperclient"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var UploadCmd = &cobra.Command{
+	Use:   "upload",
+	Short: "",
+	Long:  ``,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return mapperclient.WithClient(func(c *mapperclient.Client) error {
+			servicesIntents, err := c.ServiceIntents(context.Background(), nil)
+			if err != nil {
+				return err
+			}
+			ctxTimeout, cancel := context.WithTimeout(context.Background(), config.DefaultTimeout)
+			defer cancel()
+			intentsClient := intents.NewClientFromToken(viper.GetString(config.OtterizeAPIAddressKey), config.GetAPIToken(ctxTimeout))
+
+			discoveredIntentsToCloud := make([]intents.IntentInput, 0)
+			for _, service := range servicesIntents {
+				for _, intent := range service.Intents {
+					discoveredIntentsToCloud = append(discoveredIntentsToCloud, intents.IntentInput{
+						Client: service.Client.Name,
+						Server: intent.Name,
+						Body:   intents.IntentBody{Type: intents.IntentTypeNone},
+					})
+				}
+			}
+
+			return intentsClient.ReportDiscoveredIntents(ctxTimeout, viper.GetString(EnvIDKey), viper.GetString(SourceKey), discoveredIntentsToCloud)
+		})
+	},
+}
+
+func init() {
+	config.RegisterStringArg(UploadCmd, EnvIDKey, "environment id", true)
+	config.RegisterStringArg(UploadCmd, SourceKey, "override the source name of the intents", true)
```

Comment:
The source field should be (sometime soon) auto-filled by the CLI, with some unique identifier of the network-mapper/cluster. So that it will be by default the same source for intents that were uploaded automatically and manually.

Reply:
changed the test to "override the default source name" so it will be clearer